### ISSUE TITLE
vs2022

### ISSUE THREAD

**Issue by N1NJ4** [justcla/VSWindowManager#11]
Able to update this? Thank you!

**Comment by justcla** [comment]
I can take a look at updating it.

**Comment by justcla** [comment]
Can you please tell me, which features of the Window Manager are most important to you?
1. Status bar menu  (bottom left corner menu)
  a. Layout section
  b. Popular items section (Soln Exp, Output, Error)
  c. Most recently used items section
  d. Other - eg. Show/Hide gutters, Hide All, Open most recently closed, Fullscreen
2. New items in the Window menu
  a. Minimise all
  b. Hide most recent
  c. Close most recent
  d. Re-open [most recent]
  e. Show/Hide gutters
3. Shortcuts to Hide, Dock and Hide-All  (is there a favourite?)

I would love to know how this extension is being used, and what the most needed feature is (so that VS can build it into the product).

**Comment by N1NJ4** [comment]
I actually have never used this extension before though what drew me in was the Show/Hide Gutters. I'm interested in some of the others and trying the rest..
I think I'd find useful; Most recently used items section, Open most recently closed, Close most recent.
Primarily the Shortcuts I'm interested in, though may need a settings to be able to either find or change them if need(As unsure where to find the names of if there were no menu//items).

I may at some point reinstall 2019 though I don't know when, or if I will, then I could try it. As I'm unsure of quite the effect or even performance (not the most current device I have, so may not be worth using, though).. Minimize All.
including Hide most recent and Hide all I'd like to see in action to get a better idea of, though not a fan of unhiding.

If I'm sure of then I guess Popular items section could be included with my first list. Which is my primary and am quite interested in. More real-estate. I'm just hoping it doesn't lag so its usable at the drop of a hat.


### tldr;
Shortcuts (3.):
Show/Hide Gutters (e.)
Open most recently closed (d..)
Close most recent (c.)

Included:
Most recently used items section
AND/OR
Popular items section.
(unsure just what they would be like and if one is needed/wanted, or both)
Minimize all

Interested in, though may be less wanted(shortcuts):
Re-open [most recent]
hide all (maybe not, it is hide and I dislike unhiding things, takes too long, minimize is easy)

Only really need the shortcuts, probably need a settings regard even just to mention the names for editing the shortcuts (maybe including the key combo for reference) Could be easier than menu's short of the recently used and popular.

That being mentioned I'm very much hoping it's performant, I don't have the best device and it wouldn't be as useful if the gutters, minimize all, were causing lag, though at least there's everything else that looks useful. 

Hope both give a decent idea of my considerations and interests(despite if either list isn't the same as the other, and/or, missing regards).

I practically like the idea of most of this extension, think it could be quite useful.

**Edit: I actually don't like the idea of the status bar menu, I use a hidden taskbar so its always getting in the way when I try to use them, maybe a single toolbar button would be more useful (maybe harder though?)**

**Comment by justcla** [comment]
Thanks for all the detailed feedback!

I can respond directly to a few things.

1. If you really like the Show/Hide Gutters feature, I'm sure you'll be delighted to discover that the VS Shell team also liked this feature, and we built it into the product some time around VS2017. So if you want to show and hide the gutters, look for the following menu item:

`Window-->Show Sidebar Tabs`

2. Hide and Minimize are the same thing.

> hide all (maybe not, it is hide and I dislike unhiding things, takes too long, minimize is easy)

A window can be Open, Closed, or Hidden. Hidden is minimized. It means there is tab on the sidebar/gutter.
A window can separately be Docked(Pinned) or Undocked(Unpinned), or floating.

3. Shortcuts.
You can assign whatever shortcuts you like. It can be a bit tricky, but you'll find everything under

`Tools-->Options; Environment-->Keyboard`

You can bind shortcuts to the built-in VS commands for Hide, HideAll, Close, Dock, DockAsTabbedDocument.

`Window.Dock`   (Ctrl+Ins) - Pin the window.
`Window.AutoHide`   (Shift+Ins) - Unpin the window. Minimize to the sidebar.
`Window.AutoHideAll`     (Shift+Alt+Ins) - Minimize all windows. (There is not built-in way to restore them all.)
`Window.Close`     (Shift+Esc - This is a built-in default) - Close the window. (Not hide/minimize) Removes it from the sidebar. Window keeps its Dock state, so if it was Docked when closed, it will be Docked when open (Handy!)

Some of the commands - like CloseMostRecentWindow, ReopenLastClosedWindow, HideMostRecentWindowGroup, ToggleVisibleWindows (Show/Hide All Windows) - do not exist in Visual Studio unless this extension is installed. So you won't find them in the Keyboard settings by default, and you won't be able to assign shortcuts to them. But if you install the extension, then they will appear.

4. Thanks for the feedback about the status bar menu. Perhaps I could create an option to disable it. I like the idea of adding it to a toolbar. That would be very easy. Many of the items are also on the main Window menu, but not FullScreen (View menu), and not the Popular and Most Recent windows list. This toolbar idea is a good one. Thanks.

**Comment by N1NJ4** [comment]
Thanks, I seam to have sorted it due to your post.

**Comment by justcla** [comment]
I'm glad I could help.

I'm still working on the release for VS2022, so I'll keep the issue open.
